source/extensions/filters/http/tap/tap_config_impl.cc

```diff
@@ -0,0 +1,120 @@
+#include "extensions/filters/http/tap/tap_config_impl.h"
+
+#include "envoy/data/tap/v2alpha/http.pb.h"
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace TapFilter {
+
+HttpTapConfigImpl::HttpTapConfigImpl(envoy::service::tap::v2alpha::TapConfig&& proto_config,
+                                     Common::Tap::Sink* admin_streamer)
+    : Extensions::Common::Tap::TapConfigBaseImpl(std::move(proto_config)),
+      admin_streamer_(admin_streamer) {
+  // TODO(mattklein123): The streaming admin output sink is the only currently supported sink.
+  ASSERT(admin_streamer);
+  ASSERT(proto_config_.output_config().sinks()[0].has_streaming_admin());
```

Comment:
Should these be `RELEASE_ASSERT`s, sine they are subject to untrusted user input?

Reply:
This is verified by schema. I think regular assert is OK?

Reply:
Fair enough, but this is a bit fragile, as when we extend the proto these might become weaker than expected without noticing. Since you're the one who owns this code right now, I'm not that worried, so we can resolve this.